libs/execution_context/ebpf_program.c

```diff
@@ -2242,7 +2242,7 @@ _ebpf_program_test_run_work_item(_In_ cxplat_preemptible_work_item_t* work_item,
                 ebpf_lower_irql(old_irql);
 
                 // Reacquire the CPU.
-                old_irql = ebpf_raise_irql(context->required_irql);
+                old_irql = ebpf_raise_irql(context->maximum_irql);
```

Comment:
One issue I see with this is that the user of BPF_PROG_TEST_RUN will not be able to test a program at lower than max IRQL. Maybe have the IRQL as an optional parameter to the protocol message?

Reply:
I considered this, but the libbpf API doesn't have an option for this. Should we extend the libbpf API to include this option?

Reply:
File https://github.com/microsoft/ebpf-for-windows/issues/3157 to track this.